### PR TITLE
fix: perform more nil guard checks against LinuxProfile

### PR DESCRIPTION
**Pull request by tariq1890**

### CODE REVIEW

**Comment by jackfrancis** [comment]
/lgtm

pending unit tests

**Comment by tariq1890** [comment]
I need to change this PR. Go text/template does not short-circuit boolean operations.

https://stackoverflow.com/questions/42208440/go-template-and-function

Sigh!

File: parts/k8s/kubernetesmastercustomdata.yml [hunk 1]
```diff
@@ -201,7 +201,7 @@ write_files:
     {{WrapAsVariable "generateProxyCertsScript"}}
 {{end}}
 
-{{if HasCustomSearchDomain}}
+{{if and (HasLinuxProfile) (HasCustomSearchDomain)}}
```
Comment (jackfrancis) [inline parts/k8s/kubernetesmastercustomdata.yml]: Did we miss one here?

Comment (tariq1890) [reply parts/k8s/kubernetesmastercustomdata.yml]: yes . Thanks :)

**Comment by jackfrancis** [comment]
/lgtm

pending unit tests

**Pull request merged**